src/analysis/quaternion.lean

```diff
@@ -0,0 +1,72 @@
+/-
+Copyright (c) 2020 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+
+import data.quaternion analysis.normed_space.real_inner_product
+
+/-!
+# Quaternions as a normed algebra
+
+In this file we define the following structures on the space `ℍ := ℍ[ℝ]` of quaternions:
+
+* inner product space;
+* normed ring;
+* normed space over `ℝ`.
+
+## Notation
+
+* `ℍ` : quaternions
+
+## TODO
+
+* Define normed algebras and prove that `ℍ` form a normed division algebra over `ℝ`.
+* Prove that `ℍ` form a normed algebra over `ℂ`.
+
+## Tags
+
+quaternion, normed_ring, normed_space
+-/
+
+notation `ℍ` := ℍ[ℝ]
```

Comment:
Should this be localized notation? People sometimes use the same notation for the upper half space...
I wouldn't mind write `open_locale quaternion` to get access to this notation.